src/parser/newsparser.py

```diff
@@ -0,0 +1,47 @@
+from collections import defaultdict
+from typing import List
+
+from src.models.entities import NewsEntity
+from src.utils.util import SynonymsUtil
+
+
+class NewsParser(object):
+
+    @classmethod
+    def parse_news_by_keywords(cls, news_list: List[NewsEntity], keywords: List[str], synonyms: bool = False):
+        if synonyms:
+            return cls.__parse_news_by_keywords_and_synonyms(news_list, keywords)
+        return cls.__parse_news_by_keywords(news_list, keywords)
+
+    @classmethod
+    def __parse_news_by_keywords_and_synonyms(cls, news_list: List[NewsEntity], keywords: List[str]):
```

Comment:
It looks like this function and __parse_news_by_keywords could be merged or some part of them could be extracted.